### PR TITLE
Mat 7697 edit parameter popup

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-7697](https://jira.cms.gov/browse/MAT-7697)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/validations/ParameterSchemaValidator.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as Yup from "yup";
+
+export const ParameterSchemaValidator = Yup.object().shape({
+  parameterName: Yup.string()
+    .required("Parameter name is required")
+    .matches(
+      /^[a-zA-Z0-9\s]*$/,
+      "Only alphanumeric characters and spaces are allowed"
```
Comment (adongare) [inline src/validations/ParameterSchemaValidator.ts:8]: I think spaces are allowed

File: src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  canEdit: boolean;
+  handleApplyParameter: Function;
+  handleParameterEdit?: Function;
+  parameter?: Parameter;
+  operation?: string;
+  onClose?: Function;
+  setOpenParameterDialog?: Function;
+}
+
+export default function ExpressionBuilder({
```
Comment (adongare) [inline src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx]:
```suggestion
export default function ParameterBuilder({
```

File: src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                if (operation === "edit") {
+                  formik.setFieldValue("parameterName", "");
+                  // call handleParameterEdit
+                  onClose();
+                } else {
+                  handleApplyParameter(parameterToApply);
+                }
+              }}
+            >
+              {operation === "edit" ? "Save" : "Apply"}
```
Comment (adongare) [inline src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx]: I only see this being used for edit. is this condition required? if yes, can we use this component for apply action as well by deleting the `ParameterPane.tsx`? both look quite similar

Comment (adongare) [reply src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx]: probably we can add a tech debt to refactor this. I think we can reuse the new parameter and edit parameter forms, not much difference between them. rest looks good.

**Review by adongare — APPROVED** [review]
